### PR TITLE
UCT, UCP: Fix coverity warnings - v1.6

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Picks the fixes for coverity warnings from master to 1.6

## Why ?

Fixes potential bugs

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11605/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 5 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 06-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/b008d32fa735fe9448e18b422b436aa67666b02f/_main) | :question: ABORTED
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/def3e8a4f8e93fa762cea588b84c0be8180a0134/hpc-arm-hwi-jenkins_W3) | :question: ABORTED
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/661740d631871b45bd05dd06b13e47f7cf2a4c76/hpc-test-node-new_W0) | :question: ABORTED
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/f19fee2481a150eda72f07002ee353c79086fee1/hpc-test-node-new_W3) | :question: ABORTED
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/93664175e9c0097123c14d88c789bd5cedc43e62/hpc-test-node-gpu_W0) | :x: FAILURE
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/dfe08453fd3080f41391b517c5d3c38bec952630/hpc-test-node-gpu_W1) | :x: FAILURE
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/d4114e7b26567782decc8799353e813b92d29473/hpc-test-node-gpu_W2) | :x: FAILURE
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/119b060dbd5b1c0b813553002aa447d06a6d3fd8/hpc-test-node-gpu_W3) | :x: FAILURE
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/ec25b5cccdd666f0c1d60b99c9158b4afc930db6/hpc-test-node-new_W2) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/1773b775fbb0b95fd83de1243521b4a4e72d89aa/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/438e52bd04ef925f1c5f61de76cd80dbbe714b53/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/aaa9f42cb2339599572a1c6e29611c9edaaa12a8/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/3fc1c75229c99d85b324c57d461db4d4c386064c/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/cb32f1dfefc2f50d38a915211e949d37b09261d7/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/e7a48a093d7c528a326221f7fa7ef36f8056fb15/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/339a7f35f93021ad1fc1d18514bdb3c016c5e8a2/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/ad6fa9354dc263377728b5b516dabb41c3f5d2c3/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/af08736d598f80dd165adb686555bbd21d701fbe/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/8a73f9b3bd022e7d177446501a0135d463532d56/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/ab803c6014f52b5af559bcf56b3370ca35fc67f3/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/21851ed50fdfefd71cffe92a2667629c1fc6e455/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/9560dec327265c95597bde67f78de503f592b80f/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/b92f23eb4526a52fb16b44d173cd5f62fe1fc733/hpc-arm-hwi-jenkins_W1) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/a236880859b64499a003e32d562f892288d647de/hpc-arm-hwi-jenkins_W2) | :question: UNKNOWN
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/8bd488b715df366a8031de96819afde9fc26227c/hpc-test-node-legacy_W0) | :question: UNKNOWN
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/0dffd611d5479761116ab06ec1353275f8913040/hpc-test-node-legacy_W1) | :question: UNKNOWN
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/e0e3ff65dc527f812a8c429d74d8861210ff780a/hpc-test-node-legacy_W2) | :question: UNKNOWN
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/9a13b3b887090ba6d4f80f8cf75a7dbf41969ca8/hpc-test-node-legacy_W3) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/d19eebcf08b37f6babff5084aaf53243/raw/948f5e15283757d53a06d97ef08a9458d2c57b07/hpc-test-node-new_W1) | :question: UNKNOWN

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11610/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 06-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/d2f8b7278116f6c2b681eabc47c2e539df450bc2/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/5d479527fb0f0c71aa1bf9c0acf37d114e9f83dc/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/8dfec47fb30f29027a70f885c4088f1d18ca0a25/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/fd6edd9d8a5e02a899ca29b2447cff2f450ea23b/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/7e9a369757a4aed9b14461a1b1a0fa0474ce5d93/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/ab6c2c7e2545ff2911a38a478535cd41a818656c/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/33f1b1354201428cc599e3b72ed4d02a8ebd75c7/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/396c5bd019d7fa7a51a3b4c74db8539b4c50519c/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/27b6e2d3d85447835e7237e06a538cd061c541e5/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/32c5678c0f5bad1c14fcc345db65a480af8fe8c4/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/c5117086fd6882a194ae77f7598de49924df9c88/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/c97b920648b16432f13b62ee79eafe7d5460a483/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/593f3259b2914b44d74f9a69ba45d4e4ca937f3a/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/631ac1d0b023652a52aa28f90c712b5c47ef71c7/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/30aadb321368831acda10bcc89a0971bd1abb893/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/413b2496ee6cf77822c8391d8e12ee758b92f0be/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/a70c0a62c4246356326d1ce47888358923eaebf8/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/77ba0f06212e1d193b08e339868ceb3b0ed61807/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/3894350d97cda8fe9480339317bb83981365119f/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/7c94bfcdab9fbcbd31e331ea88f08d9172ca6179/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/9dde3939d26cc177063b5c4f20a5848efc299faa/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/58bf908052b19bf0c797c3758788208a2bda7e09/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/d106ab64c7ce1d951e0a6e536ff84fdc211efd77/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/6e0f8c76a244bbeb54f1d9beaba73c22ddc62481/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/d4dd1accd73b88da90668ca0a7c8b944abbb590c/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/3683f9245bcc073774a900a4e830e8af6b554bac/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/b52230f827615910ff84576e4a9b8444c89efc3d/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/d89a58419e5d47d5a6e8d2764875656a8562448c/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d0580a014e0bff197441abbddae9acae/raw/5546376fe96a8682e09f612b0df67f312c8a1438/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by dmitrygx** [comment]
@bureddy could you review as well, pls? this is a just copy-paste from #3964

File: src/tools/info/tl_info.c [hunk 1]
```diff
@@ -135,6 +135,7 @@ static void print_iface_info(uct_worker_h worker, uct_md_h md,
 
     if (status != UCS_OK) {
         printf("#   < failed to open interface >\n");
+        /* coverity[leaked_storage] */
```
Comment (yosefe) [inline src/tools/info/tl_info.c]: what is leaked?

Comment (dmitrygx) [reply src/tools/info/tl_info.c]: `iface`

Comment (dmitrygx) [reply src/tools/info/tl_info.c]: Coverity wrongly thinks that it could leak

**Comment by dmitrygx** [comment]
@yosefe this is HTML report from Coverity that fails for TOOLS/TL_INFO
[coverity_error.zip](https://github.com/openucx/ucx/files/3448198/coverity_error.zip)

**Comment by yosefe** [comment]
bot:pipe:retest

**Pull request merged**